grails-app/controllers/org/pih/warehouse/api/ProductSupplierApiController.groovy

```diff
@@ -28,4 +30,17 @@ class ProductSupplierApiController {
         productSupplierService.delete(params.id)
         render status: 204
     }
+
+    def create(ProductSupplierDetailsCommand productSupplierDetailsCommand) {
+        ProductSupplier productSupplier = productSupplierService.saveDetails(productSupplierDetailsCommand)
+
+        response.status = HttpStatus.CREATED.value()
+        render([data: productSupplier.toJson()] as JSON)
+    }
+
+    def update(ProductSupplierDetailsCommand productSupplierDetailsCommand) {
+        ProductSupplier updatedProductSupplier = productSupplierService.updateDetails(productSupplierDetailsCommand, params.id)
```

Comment:
And here's the update endpoint.
```
def update(ProductSupplierCommand command) {
    if (command.hasErrors()) {
        throw new ValidationException("validation errors", command.errors)
    }
    ProductSupplier productSupplier = productSupplierService.saveProductSupplier(params.id, command)

    render([data: productSupplier] as JSON)
}
```

Reply:
I still don't like the params.id as an argument to this method, but understand why it was done that way.

Reply:
Regarding the response for create and update, it is possible that we return the object as-is, but I am still working out the correct "response" for these endpoints based on best practices. There are a lot of good arguments for both sides.